docs/modules/ROOT/pages/node_pattern.adoc

```diff
@@ -409,6 +409,43 @@ has_user_data?(node, /^pass(word)?$/i)
 has_user_data?(node, ->(key) { # return true or false depending on key })
 ----
 
+== `%keyword` for named parameters
+
+Arguments can be passed as named parameters. They will be matched using `===`.
+Defaults values can be passed to `def_node_matcher` and `def_node_search`.
+
+[source,ruby]
+----
+def_node_matcher :interesting_call?, '(send _ %method ...)',
+                 method: Set[:transform_values, :transform_keys,
```

Comment:
Might be a good idea to accept arrays here as well and just implicitly convert them to sets. Seems to me this will result in nicer end user API, if it's more flexible.

Reply:
Aren't you aware of my machiavelic plan to convert the whole world to Sets? ;-)

I like that currently the API is simple, we use `<your object> === <node element>`

It would be easy to convert the values to Sets at the level of `def_node_matcher`, but it would be a bad idea to do it at the matcher call level, so I'm afraid there would be confusion...

I should add a note in the doc that `Array#===` will never work on a AST element.

Reply:
Doc [amended on `master`](https://github.com/rubocop-hq/rubocop-ast/commit/c6db1249cbdf4e0e69ede2825a7f2592b8959a44)